### PR TITLE
Windows versions

### PR DESCRIPTION
**Pull request by Vulfox**
Merge Master. 
Allows us to upgrade and downgrade a system's telegraf version.

### CODE REVIEW

**Review by rsk0 — APPROVED** [review]
LGTM

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+sudo: required
```
Comment (cthayer) [inline .travis.yml:1]: Are we using Travis?

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+environment:
```
Comment (cthayer) [inline appveyor.yml:1]: Are we using appveyor?

File: libraries/helpers.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+module Telegraf
+  module Helpers
+    def telegraf_executable
+      "\"#{::File.join(ENV['ProgramW6432'], 'telegraf', 'telegraf.exe')}\""
```
Comment (cthayer) [inline libraries/helpers.rb:4]: Is this only for windows?

File: resources/install.rb [hunk 4]
```diff
@@ -131,10 +153,22 @@
         action :create
       end
 
-      windows_zipfile ENV['ProgramW6432'] do
-        source "#{Chef::Config[:file_cache_path]}/#{file_name}"
-        not_if { ::File.exist?("#{ENV['ProgramW6432']}\\telegraf\\telegraf.exe") }
-        action :unzip
+      version_path = "#{ENV['ProgramW6432']}\\telegraf\\#{new_resource.install_version}"
+
+      unless ::File.exist?(::File.join(version_path, "telegraf", "telegraf.exe"))
```
Comment (cthayer) [inline resources/install.rb:158]: Maybe have a `executable_exists?` function in the helpers library so you aren't duplicating how paths are built?

File: resources/install.rb [hunk 5]
```diff
@@ -131,10 +153,22 @@
...
+        windows_zipfile version_path do
+          source "#{Chef::Config[:file_cache_path]}/#{file_name}"
+          not_if { ::File.exist?(::File.join(version_path, "telegraf", "telegraf.exe")) }
+          action :unzip
+        end
+      end
+
+      link "#{ENV['ProgramW6432']}\\telegraf\\telegraf.exe" do
```
Comment (cthayer) [inline resources/install.rb:166]: Same here.  A function in the helper library that returns the executable path

**Review by cthayer — COMMENTED** [review]
Couple comments/suggestions

**Pull request closed without merging**